library/src/main/java/com/google/android/exoplayer2/extractor/flv/ScriptTagPayloadReader.java

```diff
@@ -172,11 +194,18 @@ private static String readAmfString(ParsableByteArray data) {
    * @return The value read from the buffer.
    */
   private static HashMap<String, Object> readAmfEcmaArray(ParsableByteArray data) {
-    int count = data.readUnsignedIntToInt();
-    HashMap<String, Object> array = new HashMap<>(count);
-    for (int i = 0; i < count; i++) {
+    //ecma array is not the strict array, so it shouldn't be trusted number of item count
```

Comment:
Rather than this comment, it would probably be clearer to do:
int approximateCount = data.readUnsignedIntToInt();

It's probably still worth passing the value to the HashMap constructor as an initial size as well, as was previously the case.